src/hw/index.ts

```diff
@@ -77,6 +84,22 @@ export const close = (
   // fallback on an actual close
   return transport.close();
 };
+export const setAllowAutoDisconnect = (
+  transport: Transport,
+  deviceId: string,
+  allow: boolean,
+): Promise<void> => {
+  for (let i = 0; i < modules.length; i++) {
+    const m = modules[i];
+    const p =
+      m.setAllowAutoDisconnect &&
+      m.setAllowAutoDisconnect(transport, deviceId, allow);
+    if (p) return p;
+  }
+
+  // fallback on an actual close
+  return transport.close();
```

Comment:
i'm not sure about this. it changes the previous behavior: 

in the 

```
 deviceQueue
        .then(() => open(deviceId)) // open the transport
        .then(async (transport) => {
```
        
        case we were NOT doing a close(). it is important not to close() the ble transport of mobile for perf.

Reply:
No, you are absolutely right, this is not correct.
I copy pasted the code from `close` as a base and left that there. This method should only do something if the transport is implementing it, if not, it should be innocuous. Good catch